packages/gestalt/src/Masonry/MeasureItems.js

```diff
@@ -0,0 +1,69 @@
+// @flow strict
+import { Fragment, type Node, useLayoutEffect, useMemo } from 'react';
+import { type Cache } from './Cache.js';
+import { type Position } from './types.js';
+
+const layoutNumberToCssDimension = (n) => (n !== Infinity ? n : undefined);
+
+type Props<T> = {|
+  baseIndex: number,
+  getPositions: (items: $ReadOnlyArray<T>) => $ReadOnlyArray<Position>,
+  items: $ReadOnlyArray<T>,
+  measurementStore: Cache<T, number>,
+  renderItem: ({|
+    +data: T,
+    +itemIdx: number,
+    +isMeasuring: boolean,
+  |}) => Node,
+|};
+
+export default function MeasureItems<T>({
+  baseIndex,
+  getPositions,
+  items,
+  measurementStore,
+  renderItem,
+}: Props<T>): Node {
+  const measuringPositions = getPositions(items);
+  const refs = useMemo(() => new Map(), []);
+
+  useLayoutEffect(() => {
+    if (refs.size === items.length) {
+      refs.forEach((el, data) => {
+        measurementStore.set(data, el.clientHeight);
```

Comment:
would do some profiling and double check but my guess is that you'll want to do something like:
```
const heights = refs.map((el) => el.clientHeight);
heights.forEach((height) => {
  measurementStore.set(..)
});
```

this ensures that you're grouping all the calls to `clientHeight` together so the browser will batch the paints. my guess is that intermediate `measurementStore.set` calls would result in the browser triggering multiple paints still

Reply:
Ah that makes sense…that's the part I was missing from our conversation about the approach